mantidproject/mantid
# Adding markers to workbench plots

**Description of work.**
This PR adds the following to the plots in Workbench:
 - vertical and horizontal markers
 - labels for the markers
 - editing menu for label, line style and position
 - exporting markers to table workspaces
 - importing markers from table workspaces

**To test:**
Code review.

1. Open a plot in workbench
2. `right click->Markers->Horizontal/Vertical`, a green dashed line should appear with label `marker 0`
3. Add more markers, the names should increase progressively (i.e. `marker 0`, `marker 1` etc.)
4. Hover the cursor over one, the cursor should become an open hand
5. Click and drag, the cursor should become a closed hand. The label should disappear and the marker should move.
6. Release the marker, the label should appear again in the correct position
7. Pan/Zoom, the label should always be in the correct position
8. Hover the cursor over a marker and right click.
9. `marker 0->Edit`, an edit window will pop up
10. Adjust the position and change the name, the marker should move and the label should be updated
11. Right click again
12. `<name>->Delete`, marker and label should disappear
13. Now add a vertical and horizontal marker
14. Hover above the intersections and drag them. They should both move
15. Right click above the intersection. Now there should be 2 menu, one for each marker.
16. Double click a marker, the editing menu will appear.
17. The labels should not trail off the edge of the figure or lay across the marker, try using long names and labels like `$exp(-\frac{(x-\mu)^2}{2 \sigma^2})\quad \aleph$`

Fixes #26657.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
